realm/realm-library/src/main/java/io/realm/RealmList.java

```diff
@@ -150,6 +158,9 @@ public void add(int location, E object) {
         checkValidObject(object);
         if (managedMode) {
             checkValidView();
+            if (location < 0 || location > size()) {
```

Comment:
Yes, I think so, but the code just below this will copy the object to Realm before checking the index, so if you are catching the IndexOutOfBounds, you suddenly have an orphaned object. By checking first we can avoid that.